esp-hal/src/uart.rs

```diff
@@ -2141,7 +2178,11 @@ pub mod lp_uart {
                     .sclk_div_num()
                     .bits(clk_div as u8 - 1)
                     .sclk_sel()
-                    .bits(0x3) // TODO: this probably shouldn't be hard-coded
+                    .bits(match clock_source {
+                        super::ClockSource::Xtal => 3,
+                        super::ClockSource::RcFast => 2,
+                        super::ClockSource::Apb => panic!("Wrong clock source for LP_UART"),
```

Comment:
Not sure if panicking here is ok though...
Feedback is welcome, colleagues! 😄

Reply:
I think we should find a way to `cfg` it away for lp_uart